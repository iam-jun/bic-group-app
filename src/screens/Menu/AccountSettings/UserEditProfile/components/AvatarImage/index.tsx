import {
  ActivityIndicator, StyleProp, StyleSheet, View, ViewStyle,
} from 'react-native';
import React from 'react';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import images from '~/resources/images';
import Avatar from '~/baseComponents/Avatar';
import Text from '~/beinComponents/Text';
import Divider from '~/beinComponents/Divider';
import { scaleSize } from '~/theme/dimension';
import { spacing } from '~/theme';
import ButtonWrapper from '~/baseComponents/Button/ButtonWrapper';

interface Props {
  style?: StyleProp<ViewStyle>;
  url: string;
  loading: boolean;
  onPressEdit: () => void
}

const AvatarImage = ({
  style, url, loading, onPressEdit,
}: Props) => {
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = createStyles(theme);

  return (
    <View style={style}>
      <View style={styles.headerItem}>
        <Text.BodyM color={colors.neutral80} useI18n>
          settings:title_avatar
        </Text.BodyM>
        <ButtonWrapper onPress={onPressEdit} disabled={loading}>
          <Text.H6
            testID="avatar_image.edit"
            color={!loading ? colors.neutral80 : colors.gray40}
            style={styles.editBtn}
            useI18n
          >
            settings:title_edit
          </Text.H6>
        </ButtonWrapper>
      </View>
      <View style={styles.imageButton} testID="avatar_image.image">
        {!loading ? (
          <Avatar.XLarge
            style={styles.avatar}
            source={url || images.img_user_avatar_default}
            isRounded
          />
        ) : (
          <View style={[styles.avatar, styles.imageLoading]}>
            <ActivityIndicator />
          </View>
        )}
      </View>
      <Divider style={styles.divider} />
    </View>
  );
};

const createStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    avatar: {
      width: scaleSize(96),
      height: scaleSize(96),
      maxHeight: 125,
      maxWidth: 125,
      borderRadius: 8,
    },
    imageButton: {
      alignItems: 'center',
    },
    headerItem: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      paddingRight: spacing.padding.base,
      paddingVertical: spacing.padding.small,
      paddingLeft: spacing.padding.large,
      alignItems: 'center',
    },
    divider: {
      marginVertical: spacing.margin.small,
    },
    imageLoading: {
      backgroundColor: colors.gray20,
      justifyContent: 'center',
    },
    editBtn: { padding: spacing.padding.small },
  });
};

export default AvatarImage;
