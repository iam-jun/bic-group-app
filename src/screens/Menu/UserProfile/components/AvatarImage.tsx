import { ExtendedTheme, useTheme } from '@react-navigation/native';
import { StyleSheet, View } from 'react-native';

import React from 'react';
import Avatar from '~/beinComponents/Avatar';
import images from '~/resources/images';
import { spacing } from '~/theme';
import EditButton from './EditButton';

interface Props {
    id: string;
    userId: string;
    currentUsername: string;
    avatar: any;
    onEdit: () => void;
}

const AvatarImage = ({
  id, avatar, userId, currentUsername, onEdit,
}: Props) => {
  const theme:ExtendedTheme = useTheme();
  const styles = themeStyles(theme);

  return (
    <View style={styles.imageButton}>
      <View>
        <Avatar.UltraSuperLarge
          source={avatar || images.img_user_avatar_default}
          isRounded
          showBorder
        />
        <EditButton
          testID="user_profile.edit.avatar"
          style={styles.editAvatar}
          userId={userId}
          currentUsername={currentUsername}
          onPress={onEdit}
        />
      </View>
    </View>
  )
}

const themeStyles = (
  theme: ExtendedTheme,
) => {
  const { colors } = theme;

  return StyleSheet.create({
    editAvatar: {
      position: 'absolute',
      bottom: 0,
      right: spacing?.margin.small,
    },
    imageButton: {
      alignItems: 'center',
      marginTop: -44,
    },
  });
}

export default AvatarImage;
