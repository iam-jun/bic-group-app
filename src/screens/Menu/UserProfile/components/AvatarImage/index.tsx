import { ExtendedTheme, useTheme } from '@react-navigation/native';
import { StyleSheet, View } from 'react-native';

import React from 'react';
import Avatar from '~/baseComponents/Avatar';
import images from '~/resources/images';
import { spacing } from '~/theme';
import EditButton from '../EditButton';

interface Props {
  userId: string;
  currentUsername: string;
  avatar: any;
  onEdit: () => void;
}

const AvatarImage = ({
  avatar, userId, currentUsername, onEdit,
}: Props) => {
  const theme:ExtendedTheme = useTheme();
  const styles = themeStyles(theme);

  return (
    <View testID="user_profile.avatar_image" style={styles.container}>
      <View>
        <Avatar.XLarge
          source={avatar || images.img_user_avatar_default}
          isRounded
          showBorder
        />
        <EditButton
          testID="user_profile.avatar_image.edit_button"
          style={styles.editAvatar}
          userId={userId}
          currentUsername={currentUsername}
          onPress={onEdit}
        />
      </View>
    </View>
  );
};

const themeStyles = (
  theme: ExtendedTheme,
) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      alignItems: 'center',
      marginTop: -44,
    },
    editAvatar: {
      position: 'absolute',
      bottom: 0,
      right: spacing?.margin.small,
    },
  });
};

export default AvatarImage;
