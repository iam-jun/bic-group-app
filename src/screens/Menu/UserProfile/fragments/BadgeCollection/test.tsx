import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
