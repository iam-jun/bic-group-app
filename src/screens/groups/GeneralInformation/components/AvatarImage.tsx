import { ActivityIndicator, StyleSheet, View } from 'react-native';
import React from 'react';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Image from '~/beinComponents/Image';

import { useKeySelector } from '~/hooks/selector';
import groupsKeySelector from '~/storeRedux/groups/keySelector';
import images from '~/resources/images';
import spacing from '~/theme/spacing';
import InfoCard from '~/components/InfoCard';

interface Props {
  testID?: string;
  avatar: string;
  canEditInfo: boolean;

  onEditAvatar: () => void;
}

const AvatarImage = ({
  testID, avatar, canEditInfo, onEditAvatar,
}: Props) => {
  const loadingAvatar = useKeySelector(groupsKeySelector.loadingAvatar);

  const theme: ExtendedTheme = useTheme();
  const styles = themeStyles(theme);

  return (
    <InfoCard
      testID={testID}
      title="settings:title_avatar"
      onEdit={canEditInfo ? onEditAvatar : undefined}
      style={styles.container}
    >
      <View style={styles.imageButton}>
        {!loadingAvatar ? (
          <Image
            testID="avatar.image"
            style={styles.avatar}
            source={avatar || images.img_user_avatar_default}
          />
        ) : (
          <View
            testID="avatar.loading"
            style={[styles.avatar, styles.imageLoading]}
          >
            <ActivityIndicator />
          </View>
        )}
      </View>
    </InfoCard>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    container: {
      paddingBottom: spacing.padding.large,
    },
    avatar: {
      width: 145,
      height: 145,
      borderRadius: spacing.borderRadius.small,
    },
    imageButton: {
      alignItems: 'center',
    },
    imageLoading: {
      backgroundColor: colors.gray20,
      justifyContent: 'center',
    },
  });
};

export default AvatarImage;
