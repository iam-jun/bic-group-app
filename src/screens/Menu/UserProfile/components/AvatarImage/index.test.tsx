import React from 'react';

import { renderWithRedux } from '~/test/testUtils';
import AvatarImage from '.';

describe('AvatarImage component', () => {
  const baseProps = {
    isCurrentUser: true,
    avatar: null,
    onEdit: jest.fn(),
  };

  it('renders correctly', () => {
    const rendered = renderWithRedux(<AvatarImage {...baseProps} />);
    const { getByTestId } = rendered;
    const containerComponent = getByTestId('user_profile.avatar_image');
    expect(containerComponent).toBeDefined();
  });
});
