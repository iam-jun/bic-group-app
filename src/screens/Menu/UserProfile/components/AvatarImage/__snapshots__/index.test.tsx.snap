// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarImage screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": -44,
    }
  }
  testID="user_profile.avatar_image"
>
  <View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 4,
            "height": 90,
            "width": 90,
          },
          undefined,
        ]
      }
      testID="avatar"
    >
      <View
        style={
          Array [
            Array [
              Object {
                "borderRadius": 4,
                "height": 90,
                "width": 90,
              },
              Object {
                "borderRadius": 45,
              },
            ],
            Object {},
          ]
        }
        testID="avatar_container"
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "position": "relative",
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f9f9f9",
                  "borderColor": "#F2F5F8",
                  "borderRadius": 45,
                  "borderWidth": 2,
                  "height": 90,
                  "justifyContent": "center",
                  "width": 90,
                }
              }
              testID="Image__placeholder"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderColor": "#F2F5F8",
                  "borderRadius": 45,
                  "borderWidth": 2,
                  "height": 90,
                  "width": 90,
                },
              ]
            }
          >
            <FastImageView
              onFastImageError={[Function]}
              onFastImageLoadEnd={[Function]}
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../assets/images/user_avatar_default.png",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="avatar.image"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
